### PR TITLE
add query_this_hook to hooks.rs

### PR DESCRIPTION
**Pull request by ishaanh**
Added a function to hooks.rs that enables devs to check if a specific hook exists. Small but hopefully useful :)

### CODE REVIEW

File: packages/controllers/src/hooks.rs [hunk 1]
```diff
@@ -121,6 +121,19 @@ impl<'a> Hooks<'a> {
...
+    pub fn query_this_hook<Q: CustomQuery>(&self, deps: Deps<Q>, this_hook: String) -> StdResult<bool> {
+        let hooks = self.query_hooks(deps)?;
+
+        for i in 0..hooks.hooks.len() {
+             if hooks.hooks[i] == this_hook
+            {
+                return Ok(true);
+            }
+        }
+        return Ok(false);
```
Comment (ueco-jb) [inline packages/controllers/src/hooks.rs:135]:
```suggestion
        Ok(hooks.hooks.into_iter().any(|hook| hook == this_hook))
```

Comment (ueco-jb) [inline packages/controllers/src/hooks.rs:135]:
```suggestion
        Ok(hooks.hooks.into_iter().any(|hook| hook == this_hook))
```

Comment (ueco-jb) [inline packages/controllers/src/hooks.rs:135]:
```suggestion
    pub fn query_this_hook<Q: CustomQuery>(
        &self,
        deps: Deps<Q>,
        this_hook: String,
    ) -> StdResult<bool> {
        let hooks = self.query_hooks(deps)?;
        Ok(hooks.hooks.into_iter().any(|hook| hook == this_hook))
```
This fixes your lint as well.

File: packages/controllers/src/hooks.rs [hunk 2]
```diff
@@ -121,6 +121,19 @@ impl<'a> Hooks<'a> {
         let hooks = hooks.into_iter().map(String::from).collect();
         Ok(HooksResponse { hooks })
     }
+
+    // Returns true is this_hook is in hooks; false otherwiseq
```
Comment (ueco-jb) [inline packages/controllers/src/hooks.rs:125]:
```suggestion
    // Returns true is this_hook is in hooks
```
Never mind the typo, but there's no need in stating the obvious.

**Review by ueco-jb — APPROVED** [review]
LGTM
Thank you for your contribution

File: packages/controllers/src/hooks.rs [hunk 3]
```diff
@@ -121,6 +121,19 @@ impl<'a> Hooks<'a> {
         let hooks = hooks.into_iter().map(String::from).collect();
         Ok(HooksResponse { hooks })
     }
+
+    // Returns true is this_hook is in hooks
+    pub fn query_this_hook<Q: CustomQuery>(
+        &self,
+        deps: Deps<Q>,
+        this_hook: String,
```
Comment (maurolacy) [inline packages/controllers/src/hooks.rs:129]:
```suggestion
    // Returns true if hook is in hooks
    pub fn query_hook<Q: CustomQuery>(
        &self,
        deps: Deps<Q>,
        hook: String,
```

File: packages/controllers/src/hooks.rs [hunk 4]
```diff
@@ -121,6 +121,19 @@ impl<'a> Hooks<'a> {
...
+    ) -> StdResult<bool> {
+        Ok(self
+            .query_hooks(deps)?
+            .hooks
+            .into_iter()
+            .any(|hook| hook == this_hook))
```
Comment (maurolacy) [inline packages/controllers/src/hooks.rs:135]:
```suggestion
            .any(|h| h == hook))
```

**Pull request merged**